src/main/java/database/models/EntityStorageCache.java

```diff
@@ -0,0 +1,179 @@
+import java.sql.SQLException;
+import java.util.HashMap;
+
+/**
+ * @author wspride
+ */
+public class EntityStorageCache {
+    private static final String TAG = EntityStorageCache.class.getSimpleName();
+    private static final String TABLE_NAME = "entity_cache";
+
+    private static final String COL_CACHE_NAME = "cache_name";
+    private static final String COL_ENTITY_KEY = "entity_key";
+    private static final String COL_CACHE_KEY = "cache_key";
+    private static final String COL_VALUE = "value";
+    private static final String COL_TIMESTAMP = "timestamp";
+
+    private final SQLiteConnectionPoolDataSource dataSource;
+    private final String mCacheName;
+
+    public EntityStorageCache(String cacheName, SQLiteConnectionPoolDataSource dataSource) {
+        this.dataSource = dataSource;
+        this.mCacheName = cacheName;
+        Connection connection = null;
+        try {
+            connection = dataSource.getConnection();
+            execSQL(connection, getTableDefinition());
+            EntityStorageCache.createIndexes(connection);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            if (connection != null) {
+                try {
+                    connection.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    private static void execSQL(Connection connection, String query) {
+        PreparedStatement preparedStatement = null;
+        try {
+            preparedStatement = connection.prepareStatement(query);
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            if (preparedStatement != null) {
+                try {
+                    preparedStatement.close();
+                } catch (SQLException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    public static String getTableDefinition() {
+        return "CREATE TABLE " + TABLE_NAME + "(" +
+                DatabaseHelper.ID_COL + " INTEGER PRIMARY KEY, " +
+                COL_CACHE_NAME + ", " +
+                COL_ENTITY_KEY + ", " +
+                COL_CACHE_KEY + ", " +
+                COL_VALUE + ", " +
+                COL_TIMESTAMP +
+                ")";
+    }
+
+    public static void createIndexes(Connection connection) throws SQLException {
+        execSQL(connection,
+                DatabaseIndexingUtils.indexOnTableCommand("CACHE_TIMESTAMP", TABLE_NAME, COL_CACHE_NAME + ", " + COL_TIMESTAMP));
+        execSQL(connection,
+                DatabaseIndexingUtils.indexOnTableCommand("NAME_ENTITY_KEY", TABLE_NAME, COL_CACHE_NAME + ", " + COL_ENTITY_KEY + ", " + COL_CACHE_KEY));
+    }
+
+    //TODO: We should do some synchronization to make it the case that nothing can hold
+    //an object for the same cache at once
```

Comment:
i think this would be obviated with the use of redis if i understand this correctly